### PR TITLE
Docs: Line wrap long lines of text for a display

### LINKED ISSUE #1139: Read the Docs line length
Whilst looking through the docs  [here](https://opentimelineio.readthedocs.io/en/latest/tutorials/otio-serialized-schema.html?highlight=image%20sequence#imagesequencereference-1) I noticed some of the descriptive text in the documentation has a very long line length making viewing a bit of a pain. I have opened a PR to fix this issue and will have a look to see if it's the same any where else.

### PR DESCRIPTION
**Pull request by ThomasWilshaw**
**Link the Issue(s) this Pull Request is related to.**

Fix #1139 
**Summarize your change.**

The line length of some of the comments in the code blocks was extreamly long making reading it a pain. This PR manually line wraps that to 80 characters. I donlt know if there is a specific convention for this.

### CODE REVIEW

**Review by meshula — APPROVED** [review]
I think the unwrapped sources are meant to be easy to edit without having to manually rewrap, but rewrapping is easy, so I'm personally fine with checking it in wrapped.

**Comment by ssteinbach** [comment]
This documentation is actually autogenerated, by extracting from pydoc directly from the classes.  So the line breaking would have to happen in this script:

https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/27ed98663fb4e20b9e187f85b34f5a58f9bd60d9/src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py#L319

**Review by meshula — CHANGES_REQUESTED** [review]
Switching to Request Changes, per @ssteinbach's note.

**Comment by ThomasWilshaw** [comment]
Might it be better to wrap the lines [here](https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/3e4b4591409a4afce92ca7b2bc9cb94c7f756426/src/py-opentimelineio/opentimelineio-bindings/otio_serializableObjects.cpp#L702) rather than trying to do it automatically and risking wrapping very long commands that should probably stay on a single line?

**Comment by ssteinbach** [comment]
Thats an option.  There are also tools in python for wrapping text:
https://docs.python.org/3/library/textwrap.html

@meshula I'm a little surprised our clang format pass didn't wrap these lines, actually.

**Comment by JeanChristopheMorinPerso** [comment]
@ssteinbach those string are prefixed with `R`, which I guess it's kind of a raw string. It would be weird if clang formatted these raw string no?

**Comment by ThomasWilshaw** [comment]
How does that look? It's been a while since Ive written any Python code so there may be a nicer/better way to do it.

**Comment by ssteinbach** [comment]
The unit tests check to see if the autogenerated docs have been updated -- you'll need to check in to your branch the updated docs.  You can generate them by using the `make doc-model-update` build target.

**Comment by ThomasWilshaw** [comment]
Done, but for some reason there has been a change in the text that's nothing to do with the Python code

**Comment by meshula** [comment]
For certain, we would never want clang-format to interfere with an R"string", as that would interfere with the very purpose of the raw strings.

**Comment by ThomasWilshaw** [comment]
I've just updated this. I had an older version of OTIO in my Python setup that was messing things up

**Review by meshula — APPROVED** [review]

**Comment by meshula** [comment]
@ThomasWilshaw Sorry to have missed your request for a review, I just spotted this. Now, there is the CLA to take care of. If you're able to go through the CLA business, we can get this PR moving again. Apologies again.

**Comment by ThomasWilshaw** [comment]
No problem, sorry I've dropped off the radar for a while, I've been very busy of late. That's the CLA all done, please let me know if there are any other issues.

**Comment by meshula** [comment]
Looks like CI is stumbling as soon as it hits the py_build_test.  My python-fu is too weak ~ any thoughts on what's tripping it up?

File: src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py
Comment (meshula) [inline src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py:303]:
```suggestion
            if(cl.__doc__ is not None):
```

File: src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py
Comment (meshula) [inline src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py:307]:
```suggestion
                    line = textwrap.wrap(
                                    line, width=100, 
                                     expand_tabs=False, 
                                     replace_whitespace=False, 
                                     drop_whitespace=False, 
                                     break_long_words=False)
```

File: src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py
Comment (meshula) [inline src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py:312]:
```suggestion
                    line = textwrap.wrap(line, width=100,
                                         expand_tabs=False,
                                         replace_whitespace=False,
                                         drop_whitespace=False,
                                         break_long_words=False)
```

File: src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py
Comment (meshula) [inline src/py-opentimelineio/opentimelineio/console/autogen_serialized_datamodel.py:314]:
```suggestion
                    for wrapped_line in line:
                        new_docstring.append(wrapped_line)
```

**Comment by meshula** [comment]
@ThomasWilshaw I corrected the PEP8 issues that came up, but it looks like there needs to be a corresponding update to the test suite to accommodate the reformatting?

**Comment by JeanChristopheMorinPerso** [comment]
I believe `make doc-model-update` should do the trick.

**Comment by jminor** [comment]
It looks like this PR is almost ready. It just needs `make doc-model-update` and DCO signoff.

**Comment by ThomasWilshaw** [comment]
Apologies for the massive massive delay on completing this but I think it is all good to go now aside form the DCO stuff. Would someone be able to advise on how I'm supposed to sort that?

**Comment by meshula** [comment]
The DCO requirement would have needed you to do all your commits with the `-s` argument. That would be tedious to retrofit. Instead, when you are ready to merge, we'll do a squash merge after forcing the DCO to pass. If you could do one last commit, with -s, it'll be easy enough for us to force that DCO signoff to apply to whole thing.

**Comment by JeanChristopheMorinPerso** [comment]
BTW, you can see the command line to use at https://github.com/AcademySoftwareFoundation/OpenTimelineIO/runs/10531037992.

**Comment by jminor** [comment]
It is pretty easy to retroactively sign-off all of the commits like this:
```
% git status
...check that you're on the right branch ("line_wrap" in your case)
% git rebase main --signoff
...check that the history looks fine (code is unchanged by this command)
% git push --force
```

The signoff step is intended to make sure you've checked that all of the commits are what you've intended to contribute, and nothing else accidentally got checked in.

**Comment by jminor** [comment]
Oops, it looks like we're all helping to answer at the same time. I didn't see @meshula 's or @JeanChristopheMorinPerso 's advice before posting. Luckily these are all good, mostly overlapping, options. If the commands above don't work for you, let us know and we can do the step that @meshula suggested on your behalf.

**Comment by meshula** [comment]
@joshm TIL a non-tedious way to fix the sign-off requirement ;)

**Comment by meshula** [comment]
> I should signoff any and all commits to this repo

Yes, I have to remind myself as well, to add a `-S` argument to commits.

**Pull request merged**
